lib/src/wiggle_abi/dictionary_impl.rs

```diff
@@ -6,23 +6,62 @@ use {
         session::Session,
         wiggle_abi::{fastly_dictionary::FastlyDictionary, types::DictionaryHandle},
     },
+    std::{
+        convert::{
+            TryFrom,
+            TryInto,
+        },
+        fs,
+        path::Path
+    },
     wiggle::GuestPtr,
 };
 
+fn read_json_file<P: AsRef<Path>>(file: P) -> serde_json::Map<String, serde_json::Value> {
+    let data = fs::read_to_string(file).expect("Unable to read file");
+    let json: serde_json::Value = serde_json::from_str(&data).expect("JSON was not well-formatted");
+    let obj = json.as_object().expect("Expected the JSON to be an Object");
+    obj.clone()
+}
+
 impl FastlyDictionary for Session {
-    #[allow(unused_variables)] // FIXME: Remove this directive once implemented.
     fn open(&mut self, name: &GuestPtr<str>) -> Result<DictionaryHandle, Error> {
-        Err(Error::NotAvailable("Dictionary lookup"))
+        let name = name.as_str()?.to_owned();
+        self.dictionary_handle(&name)
     }
 
-    #[allow(unused_variables)] // FIXME: Remove this directive once implemented.
     fn get(
         &mut self,
         dictionary: DictionaryHandle,
         key: &GuestPtr<str>,
         buf: &GuestPtr<u8>,
         buf_len: u32,
     ) -> Result<u32, Error> {
-        Err(Error::NotAvailable("Dictionary lookup"))
+        let key = key.as_str()?.to_owned();
```

Comment:
FWIW, I think you can skip the `to_owned()` here (to avoid a copy), similar to the comment above.

Reply:
I couldn't fully avoid the copy in this function for key but, I could limit the copy to only happen when the key does not exist within the dictionary -- [`b6c2e4e` (#61)](https://github.com/fastly/Viceroy/pull/61/commits/b6c2e4ea4b5ff9cc0354095e0544b80a5f1d1bfd)